pkgs/by-name/de/deltatouch/package.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, stdenv
+, fetchFromGitea
+, cmake
+, intltool
+, libdeltachat
+, qt5
+, quirc
+, lomiri
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deltatouch";
+  version = "unstable";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "lk108";
+    repo = "deltatouch";
+    rev = "1943cb60ce7496b690936de512b30d9c5f626926";
+    hash = "sha256-SCbhnIM97GeshohqHLRWWl+G9FXOYMk5+RcpSa28uts=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    qt5.wrapQtAppsHook
+    intltool
+    cmake
+  ];
+
+  patchPhase = ''
+    mkdir -p $out/share
+    mkdir -p $out/lib
+
+    substituteInPlace CMakeLists.txt --replace "set(DATA_DIR /)" "set(DATA_DIR $out/share)"
+    substituteInPlace plugins/DeltaHandler/CMakeLists.txt --replace '/lib/' '${placeholder "out"}/lib/'
+    substituteInPlace plugins/DTWebEngineProfile/CMakeLists.txt --replace '/lib/' '${placeholder "out"}/lib/'
+    mkdir -p build/plugins/DeltaHandler
+    cp --no-preserve=mode ${libdeltachat}/lib/libdeltachat.so build/plugins/DeltaHandler/libdeltachat.so
+    cp --no-preserve=mode ${quirc}/lib/libquirc.so.1.0 build/plugins/DeltaHandler/libquirc.so.1.2
+  '';
+
+  postInstall = ''
+    mkdir -p $out/bin
+    mv $out/deltatouch $out/bin
+    cp --no-preserve=mode ${libdeltachat}/lib/libdeltachat.so $out/lib/libdeltachat.so
+    cp --no-preserve=mode ${quirc}/lib/libquirc.so.1.0 $out/lib/libquirc.so.1.2
+  '';
+
+  qtWrapperArgs = [
+    ''--prefix QML2_IMPORT_PATH : ${placeholder "out"}/lib''
+    ''--set CLICKABLE_DESKTOP_MODE y'' # Hack to make sending/receiving messages work, we don't use clickable
+  ];
+
+  buildInputs = [
+    qt5.qtbase
+    qt5.qtwebengine
+    qt5.qtquickcontrols2
+    lomiri.lomiri-ui-toolkit
+    lomiri.lomiri-ui-extras
+    lomiri.lomiri-api
+    lomiri.lomiri-indicator-network # Lomiri.Connectivity module
+    lomiri.qqc2-suru-style
+    lomiri.suru-icon-theme
```

Comment:
I think icons are one of the few things we want to be impurely-supplied to the environment, so different icon themes can be used without rebuilding everything.

Reply:
That's all QML modules, so should stay.